editor/editor_undo_redo_manager.cpp

```diff
@@ -150,6 +150,7 @@ void EditorUndoRedoManager::create_action(const String &p_name, UndoRedo::MergeM
 	if (p_custom_context) {
 		// This assigns history to pending action.
 		get_history_for_object(p_custom_context);
+		force_fixed_history();
```

Comment:
`force_fixed_history()` is supposed to be called manually. EditorUndoRedoManager will automatically deduce optimal history, if you force a different one it might lead to bugs. The method should be used if automatic history fails (there are some cases where it can't be determined), but you know that the forced one is correct.

Reply:
Without a change similar to this one, the `custom_context` parameter of `create_action` does not always function correctly (or maybe it never functions correctly). If you want, I can open a new issue describing this in more detail?

Reply:
Feel free to open it, but it might be misunderstanding.

Reply:
I've filed an issue to better describe the issue that this PR aims to resolve: #100108